components/heap/heap_trace_standalone.c

```diff
@@ -167,52 +199,71 @@ void heap_trace_dump_caps(const uint32_t caps) {
 
 static void heap_trace_dump_base(bool internal_ram, bool psram)
 {
+    // In order to keep the linked list valid,
+    // we must stop tracing
+    bool was_tracing = false;
+    if (tracing) {
```

Comment:
perhaps:
```
bool was_tracing = tracing;
if (was_tracing) {
  tracing = false;
}
```
later the code can simply assign `tracing = was_tracing` and avoid another if check.

Reply:
ya thats cleaner!

Reply:
On second look, since I already grab the mutex, `portENTER_CRITICAL(&trace_mux);`, I don't need to stop tracing. And stopping tracing means some records will be skipped.

It will cause all other code to stop executing for a long time though. Hard to say it that's "good". But I think it's preferable since the user can always stop tracing themself.

Reply:
updated.